Add blogpost for pydantic v2 (#2492)
home-assistant/developers.home-assistant: blog/2024-12-10-pydantic.md

---
author: Joost Lekkerkerker
authorURL: https://github.com/joostlek
authorImageURL: https://avatars.githubusercontent.com/u/7083755?v=4

> title: "Pydantic bump to v2"

Reviewer: ```suggestion
title: "Pydantic v2 mandatory from Home Assistant Core 2025.1"
```
Maybe something like this?

Author: `mandatory` sounds like everything should use it, which is incorrect, it's only for pydantic v1-based intergation

Author: But I do agree the current one is subpar, so open for other suggestions. We should both be trying to be a reason for action, but also make sure we target the right people

Reviewer: Indeed, poor choice of words.. How about  "Dropping support for pydantic v1 in 2025.1"?

Author: "Moving from Pydantic v1 to v2"? (in 2025.1)?

Reviewer: Sounds also good, I'd keep the version somehow in the title as it's a clear indication when the change happens.

---

Pydantic is a widely used library in Python for data validation. On June 30, 2023, Pydantic v2 was released, introducing significant changes that are not backward compatible with Pydantic v1.

> Home Assistant Core 2024.12 continues to ship with the Pydantic v1 library. Over the past year, our community has worked hard to ensure that the libraries used by Home Assistant Core are compatible with both Pydantic v1 and v2. This dual compatibility has helped make our transition to Pydantic v2 as smooth as possible.

Reviewer: ```suggestion
Over the past year, our community has worked hard to ensure that the libraries used by Home Assistant Core are compatible with both Pydantic v1 and v2. This dual compatibility has helped make our transition to Pydantic v2 as smooth as possible.
```
This reads a bit off, perhaps it's not necessary at ll?

Author: We still have a few subreleases until the end of the year, so I think it might help being explicit here that this only applies from 2025.1 onwards

> Starting with Home Assistant Core 2025.1, Pydantic v2 will replace v1. If your custom integration uses Pydantic, it must be updated to support Pydantic v2 to keep working in the upcoming release.

Reviewer 2: Swap the order of paragraph 2 and 3? This gets to the action needed sooner on what action needs to happen.

---
author: Joost Lekkerkerker
authorURL: https://github.com/joostlek
authorImageURL: https://avatars.githubusercontent.com/u/7083755?v=4
title: "Pydantic bump to v2"
---

Pydantic is a popular library for data validation in Python.
30 june 2023, Pydantic v2 was released.
This release is not backwards compatible with Pydantic v1.

Home Assistant Core 2024.12 ships with the Pydantic v1 library.
Over the last year, the community has been working on making sure the libraries used by Home Assistant Core are supporting both v1 and v2 of Pydantic.

> By supporting both versions, we tried to make our transition to Pydantic v2 as smooth as possible.

Reviewer 3: ```suggestion
By supporting both versions, we made our transition to Pydantic v2 as smooth as possible.
```

From 2025.1, Home Assistant Core is expected to ship with Pydantic v2.

> This means that every custom component that uses Pydantic will need to be updated to support Pydantic v2 in order to keep working in Home Assistant Core 2025.1.

Reviewer 3: ```suggestion
Every custom component that uses Pydantic must be updated to support Pydantic v2 to keep working in Home Assistant Core 2025.1.
```

---
author: Joost Lekkerkerker
authorURL: https://github.com/joostlek
authorImageURL: https://avatars.githubusercontent.com/u/7083755?v=4
title: "Pydantic bump to v2"
---

> Pydantic is a popular library for data validation in Python.
> 30 june 2023, Pydantic v2 was released.
> This release is not backwards compatible with Pydantic v1.

Reviewer 4: ```suggestion
Pydantic is a widely used library in Python for data validation. On June 30, 2023, Pydantic v2 was released, introducing significant changes that are not backward compatible with Pydantic v1.
```

> Home Assistant Core 2024.12 ships with the Pydantic v1 library.
> Over the last year, the community has been working on making sure the libraries used by Home Assistant Core are supporting both v1 and v2 of Pydantic.
> By supporting both versions, we made our transition to Pydantic v2 as smooth as possible.

Reviewer 4: ```suggestion
Home Assistant Core 2024.12 continues to ship with the Pydantic v1 library. Over the past year, our community has worked hard to ensure that the libraries used by Home Assistant Core are compatible with both Pydantic v1 and v2. This dual compatibility has helped make our transition to Pydantic v2 as smooth as possible.
```

From 2025.1, Home Assistant Core is expected to ship with Pydantic v2.

> Every custom component that uses Pydantic must be updated to support Pydantic v2 to keep working in Home Assistant Core 2025.1.

Reviewer 3: ```suggestion
Every custom component that uses Pydantic must be updated to support Pydantic v2 or use the v1 backward compatibility namespace to keep working in Home Assistant Core 2025.1.
```

Author: Oh I just added the note about the migration with the shims

> From 2025.1, Home Assistant Core is expected to ship with Pydantic v2.
> Every custom component that uses Pydantic must be updated to support Pydantic v2 to keep working in Home Assistant Core 2025.1.

Reviewer 4: ```suggestion
Starting with Home Assistant Core 2025.1, Pydantic v2 will replace v1. If your custom integration uses Pydantic, it must be updated to support Pydantic v2 to keep working in the upcoming release.
```
